src/controller/controller.ts

```diff
@@ -509,8 +509,13 @@ class Controller extends events.EventEmitter {
             // Reponse to time reads
             if (frame.isGlobal() && frame.isCluster('genTime') && frame.isCommand('read')) {
                 const time = Math.round(((new Date()).getTime() - OneJanuary2000) / 1000);
+                const localTime = time + (new Date()).getTimezoneOffset() * 60;
                 try {
-                    await endpoint.readResponse(frame.Cluster.ID, frame.Header.transactionSequenceNumber, {time});
+                    await endpoint.readResponse(frame.Cluster.ID, frame.Header.transactionSequenceNumber, {
```

Comment:
Although the attributes are mandatory, it doesn't mean they always have to be returned.

In order to determine what attributes need to be responded to, please check `dataPayload.frame.Payload` (it contains the attributes the device wants to read.

Reply:
Done, please take another look.

Reply:
Looks good, I will add test cases and merge it.